eclass/linux-mod-r1.eclass

```diff
@@ -461,16 +465,50 @@ linux-mod-r1_src_install() {
 	einstalldocs
 }
 
+# @FUNCTION: linux-mod-r1_pkg_preinst
+# @DESCRIPTION:
+# Find kernel version of built modules
+linux-mod-r1_pkg_preinst() {
```

Comment:
Would rather not introduce a preinst on a whim, also if wanted to know the version it was built for there is no need to do any of this. Just need to store KV_DIR and KV_FULL to a different variable that linux-info won't reset, could be done in `_modules_prepare_kernel` given it's not ran for binary packages.

Can use MODULES_GLOBAL to avoid polluting namespace too, e.g.
```
    [[ -n ${KV_FULL} ]] ||
        die "kernel sources are required to build kernel modules"

	_MODULES_GLOBAL[built:kv_dir]=${KV_DIR}
	_MODULES_GLOBAL[built:kv_full]=${KV_FULL}
	_MODULES_GLOBAL[built:kv_out_dir]=${KV_OUT_DIR}
```
If wonder about the 3rd line it's because `_modules_update_depmod` needs to replace kv_full and kv_out_dir as well.

Do have mixed feelings about no longer being able to trust linux-info's variables but well, it's not super invasive to do this.

Reply:
> trust linux-info's variables but well, it's not super invasive to do this
On that note should probably add a
```
	# preserve linux-info variables re-used in pkg_postinst due to bug #931213
```
Or else it looks kind of odd.

Reply:
> Would rather not introduce a preinst on a whim, also if wanted to know the version it was built for there is no need to do any of this. Just need to store KV_DIR and KV_FULL to a different variable that linux-info won't reset, could be done in `_modules_prepare_kernel` given it's not ran for binary packages.

Thanks, this is a lot smarter and simpler then what I was doing.

> Do have mixed feelings about no longer being able to trust linux-info's variables but well, it's not super invasive to do this.

I have dropped this part from this PR for now. Anything we do in `linux-mod-r1` we would then also have to do in `linux-firmware` and `intel-microcode`, so it might make more sense to fix it at the root of the problem (i.e. `linux-info` or `dist-kernel-utils`). I'll dig a bit deeper into this, and maybe revisit this in a separate commit later.

Reply:
> I have dropped this part from this PR for now. Anything we do in linux-mod-r1 we would then also have to do in linux-firmware and intel-microcode

Actually, no, I was wrong. We don't need such a fix in linux-firmware or intel-microcode. This problem is only relevant when the binpkg contains stuff that has been compiled for a specific kernel version (i.e. a kernel module). Therefore it is only a problem for `linux-mod-r1`.

Reply:
I've been digging into this a bit, and I think I've found a rather elegant solution. We can simply set `LINUX_INFO_BINARY_RESET=1` in `linux-mod-r1.eclass` and this will make `linux-info.eclass` not perform the problematic unsetting.

Reply:
fwiw it's an undocumented variable though, aka relying on internals

Reply:
Right, but that is a problem we can easily fix by adding some documentation :P

Reply:
(not that a documented variable couldn't be added to linux-info so both ebuilds and eclasses could freely set it)

Reply:
CC @floppym @gentoo/kernel for `linux-info` changes